### ISSUE TITLE
How to setup gradient_accumulation?

### ISSUE THREAD

**Issue by piegu** [huggingface/setfit#173]
Hi,

in order to train a model SetFit, I would like simulate a `batch_size` of 16 but with a `batch_size` of 8. For doing that, I need to setup `gradient_accumulation` to 2. 

How to do that?

Thanks.

**Comment by tomaarsen** [comment]
I'm afraid that gradient accumulation is not supported at this time. Hopefully this will be implemented some time soon. I suspect the main bottleneck for this is that [SentenceTransformers](https://github.com/UKPLab/sentence-transformers), on which SetFit relies heavily, does not implement gradient accumulation in its [`fit`](https://www.sbert.net/docs/package_reference/SentenceTransformer.html#sentence_transformers.SentenceTransformer.fit) method.

I hope this answers your question somewhat. I'll close this for the time being.
